WordPress/Classes/Categories/NSString+Helpers.m

```diff
@@ -126,9 +139,37 @@ - (NSString *)stringByReplacingHTMLEmoticonsWithEmoji
             [result replaceCharactersInRange:[match range] withString:replacement];
         }
     }
+
+    matches = [coreEmojiRegex matchesInString:result options:0 range:NSMakeRange(0, [result length])];
+    for (NSTextCheckingResult *match in [matches reverseObjectEnumerator]) {
+        NSRange range = [match rangeAtIndex:1];
+        NSString *filename = [result substringWithRange:range];
+        NSString *replacement = [self emojiUnicodeFromCoreEmojiFilename:filename];
+        if (replacement) {
+            [result replaceCharactersInRange:[match range] withString:replacement];
+        }
+    }
+
     return [NSString stringWithString:result];
 }
 
+/**
+ Pass the hex string for a unicode character and returns that unicode character. 
+ Core WordPress emoji images are named after their unicode code point. This makes
+ it convenient to scan the filename for its hex value and return the appropriate
+ emoji character.
+ */
+- (NSString *)emojiUnicodeFromCoreEmojiFilename:(NSString *)filename
+{
+    NSScanner *scanner = [NSScanner scannerWithString:filename];
+    unsigned long long hex = 0;
+    BOOL success = [scanner scanHexLongLong:&hex];
+    if (!success) {
+        return nil;
+    }
+    return [[NSString alloc] initWithBytes:&hex length:4 encoding:NSUTF32LittleEndianStringEncoding];
```

Comment:
As we discussed in Slack, it'd be nice to check that the result is an emoji character, just to be safe.